### PR TITLE
Improvement/txs sets

### PR DESCRIPTION
**Pull request by vkatsuba**

### CODE REVIEW

File: apps/arweave/src/ar_node_worker.erl [hunk 1]
```diff
@@ -314,15 +314,15 @@ move_tx_to_mining_pool(StateIn, TX, GS) ->
...
 		{gossip, NewGS},
-		{waiting_txs, WaitingTXs -- [TX]}
+		{waiting_txs, sets:del_element(TX, WaitingTXs)}
 	]}.
 
 drop_waiting_txs(#{ waiting_txs := WaitingTXs }, DropTXs, GS) ->
 	{NewGS, _} = ar_gossip:send(GS, {drop_waiting_txs, DropTXs}),
 	DropSet = sets:from_list(lists:map(fun(TX) -> TX#tx.id end, DropTXs)),
-	FilteredTXs = lists:filter(
+	FilteredTXs = sets:filter(
```
Comment (ldmberman) [inline apps/arweave/src/ar_node_worker.erl]: We could simplify it as `sets:subtract(WaitingTXs, sets:from_list(DropTXs))`

Comment (ldmberman) [inline apps/arweave/src/ar_node_worker.erl]: We could simplify it as `sets:subtract(WaitingTXs, sets:from_list(DropTXs))`

Comment (ldmberman) [inline apps/arweave/src/ar_node_worker.erl]: We could simplify it as `sets:subtract(WaitingTXs, sets:from_list(DropTXs))`.

Comment (vkatsuba) [reply apps/arweave/src/ar_node_worker.erl]: Done.

**Comment by ldmberman** [comment]
[Cherry-picked it](https://github.com/ArweaveTeam/arweave/commit/813e6783c12e5133d1c42afb88ba33405d2b8c94), thank you!

**Pull request closed without merging**